tools/toolchains/xcode_configure/xcode_configure.bzl

```diff
@@ -275,6 +286,7 @@ xcode_autoconf = repository_rule(
     attrs = {
         "xcode_locator": attr.string(),
         "remote_xcode": attr.string(),
+        "explicit_modules": attr.bool(),
```

Comment:
How long does it take to run this one? This will run many times per day having the requirement of being a `local` rule.  Longer term, maybe we'd breaking the `xcode_locator` into it's [own repo rule](https://github.com/bazel-ios/rules_ios/issues/484#issuecomment-1221169347) which will allow us to add non local cache semantics on the pcm gem::
- `xcode_locator(local=True)`
- `pcm_gem(sdk=X, xcode_locator, local=False)`

Reply:
It takes about a minute to run through all the SDKs in 3 Xcode versions from a clean/expunged repo.

Reply:
That's pretty good perf! Maybe we can decompose it for Xcode 14 stuff. They wrote a nice description about [recommending local=False](https://blog.bazel.build/2017/02/22/repository-invalidation.html#declare-your-repository-as-local-very-carefully) a few years ago BTW

Reply:
I'll plan on splitting this up a bit more to avoid the issues with `local`. If we have to run the xcode locator twice, so be it, but it'd be nice to kill two birds with one stone. If nothing else, I can at least refactor this to reuse the same code.

On top of all this, I'm still not 100% married to the idea of keeping this as something we generate on the fly every time. Assuming the SDKs are fairly stable within major Xcode versions or across major OS versions for the respective platforms, it might be as simple as generating _enough_ of them to get good coverage and making it easy to update for future Xcode/platform releases. I'd need to download a few more versions of Xcode to test though ⏳

Reply:
> Assuming the SDKs are fairly stable within major Xcode versions or across major OS versions for the respective platforms, it might be as simple as generating enough of them 

Recently we've seen updates moving fast: shipping later updates in minor releases add new frameworks or dependencies quite often - or even from `13.3` -> `13.3.1` - with the M1 perf it should be a fair tradeoff 😅 

We can always circle back to it at a much later point as well - or do it orthogonally! I imagined the perf of genning these repos on the fly is a drop in the bucket compared to the wins we get